packages/shared/src/hooks/useFeedLayout.ts

```diff
@@ -0,0 +1,33 @@
+import { useFeature } from '../components/GrowthBookProvider';
+import { feature } from '../lib/featureManagement';
+import { FeedLayout } from '../lib/featureValues';
+import { AllFeedPages } from '../lib/query';
+import { SharedFeedPage } from '../components/utilities';
+
+interface UseFeedLayoutProps {
+  feedName: AllFeedPages;
+}
+
+interface UseFeedLayoutReturn {
+  isSingleColumnFeedLayout: boolean;
+}
+
+export const useFeedLayout = ({
+  feedName,
+}: UseFeedLayoutProps): UseFeedLayoutReturn => {
+  const feedLayoutVersion = useFeature(feature.feedLayout);
+  const isFeedLayoutVersion = feedLayoutVersion === FeedLayout.V1;
+  const isIncludedFeed = Object.values(SharedFeedPage).includes(
+    feedName as SharedFeedPage,
+  );
+  const shouldUseFeedLayout = isFeedLayoutVersion && isIncludedFeed;
+
+  if (shouldUseFeedLayout) {
+    return {
+      isSingleColumnFeedLayout: true,
+    };
+  }
+  return {
+    isSingleColumnFeedLayout: false,
+  };
```

Comment:
I think this logic can be simplified to a lot less lines (maybe even one, if we inline the `isSingleColumnFeedLayout` variable):
```suggestion
  const isSingleColumnFeedLayout = isFeedLayoutVersion && isIncludedFeed;
  return { isSingleColumnFeedLayout };
```

Reply:
Yeah, based on Ante's comment as well will change this, my mind was set on separating the cases because other props might not be only related to the bool value of `isFeedLayoutVersion && isIncludedFeed`, but will change based on suggestions as it definitely makes sense now :D Thank u